entrypoint.sh

```diff
@@ -0,0 +1,122 @@
+
+    export DETECTED_DB_BACKEND=""
+    export DETECTED_DB_USER=""
+    export DETECTED_DB_HOST=""
+    export DETECTED_DB_PORT=""
+    export DETECTED_DB_SCHEMA=""
+    export DETECTED_DB_PARAMS=""
+
+    if [[ ${DB_URL} != sqlite* ]]; then
+        # Auto-detect DB parameters
+        [[ ${DB_URL} =~ ([^:]*)://([^@/]*)@?([^/:]*):?([0-9]*)/([^\?]*)\??(.*) ]] && \
+            DETECTED_DB_BACKEND=${BASH_REMATCH[1]} &&
+            DETECTED_DB_USER=${BASH_REMATCH[2]} &&
+            DETECTED_DB_HOST=${BASH_REMATCH[3]} &&
+            DETECTED_DB_PORT=${BASH_REMATCH[4]} &&
+            DETECTED_DB_SCHEMA=${BASH_REMATCH[5]} &&
+            DETECTED_DB_PARAMS=${BASH_REMATCH[6]}
+
+        echo DB_BACKEND="${DB_BACKEND:=${DETECTED_DB_BACKEND}}"
+
+        if [[ -z "${DETECTED_DB_PORT}" ]]; then
+            if [[ ${DB_BACKEND} == "postgres"* ]]; then
+                DETECTED_DB_PORT=5432
+            elif [[ ${DB_BACKEND} == "mysql"* ]]; then
+                DETECTED_DB_PORT=3306
+            fi
+        fi
+
+        DETECTED_DB_HOST=${DETECTED_DB_HOST:="localhost"}
+
+        # Allow the DB parameters to be overridden by environment variable
+        echo DB_HOST="${DB_HOST:=${DETECTED_DB_HOST}}"
+        echo DB_PORT="${DB_PORT:=${DETECTED_DB_PORT}}"
+
+        while true
+        do
+            set +e
+            LAST_CHECK_RESULT=$(nc -zvv "${DB_HOST}" "${DB_PORT}" >/dev/null 2>&1)
+            RES=$?
+            set -e
+            if [[ ${RES} == 0 ]]; then
+                echo
+                break
+            else
+                echo -n "."
+                DB_CHECK_MAX_COUNT=$((DB_CHECK_MAX_COUNT-1))
+            fi
+            if [[ ${DB_CHECK_MAX_COUNT} == 0 ]]; then
+                echo
+                echo "ERROR! Maximum number of retries (${DB_CHECK_MAX_COUNT}) reached while checking ${DB_BACKEND} db. Exiting"
+                echo
+                break
+            else
+                sleep "${DB_CHECK_SLEEP_TIME}"
+            fi
+        done
+        if [[ ${RES} != 0 ]]; then
+            echo "        ERROR: ${BACKEND} db could not be reached!"
+            echo
+            echo "${LAST_CHECK_RESULT}"
+            echo
+            export EXIT_CODE=${RES}
+        fi
+    fi
+}
+
+# if no DB configured - use sqlite db by default
+AIRFLOW__CORE__SQL_ALCHEMY_CONN="${AIRFLOW__CORE__SQL_ALCHEMY_CONN:="sqlite:///${AIRFLOW_HOME}/airflow.db"}"
+
+verify_db_connection "${AIRFLOW__CORE__SQL_ALCHEMY_CONN}"
+
+AIRFLOW__CELERY__BROKER_URL=${AIRFLOW__CELERY__BROKER_URL:=}
+
+if [[ -n ${AIRFLOW__CELERY__BROKER_URL} ]] && \
+        [[ ${AIRFLOW_COMMAND} =~ ^(scheduler|worker|flower)$ ]]; then
+    verify_db_connection "${AIRFLOW__CELERY__BROKER_URL}"
+fi
+
+if [[ ${AIRFLOW__CORE__SQL_ALCHEMY_CONN} == sqlite* ]] && \
+        [[ ${AIRFLOW_COMMAND} =~ ^(scheduler|worker|webserver)$ ]] || \
+        [[ ${DB_RESET:="false"} == "true" ]]; then
+    if [[ ${AIRFLOW_VERSION} == *1.10* ]]; then
```

Comment:
This could match 2.1.10 in future

Reply:
Aaaaargh!

Reply:
Good catch @ashb

Reply:
Indeed. Good one. I would still want to keep it as I would like to use the same Dockerfile/entrypoint.sh in both master and v1-10-test without having to change anything when backporting. This line allows that.